test/model_simulation/simulate_SDEs.jl

```diff
@@ -121,29 +121,59 @@ end
 
 ### Noise Scaling ###
 
-# Tests with a single noise scaling parameter.
+# Tests with multiple noise scaling parameters directly in the macro.
+let 
+    noise_scaling_network_1 = @reaction_network begin 
+        @noise_scaling_parameters η1 η2
+        (k1, k2), X1 ↔ X2 
+    end
+    u0 = [:X1 => 1000.0, :X2 => 3000.0]
+    sol_1_1 = solve(SDEProblem(noise_scaling_network_1, u0, (0.0, 1000.0), [:k1 => 2.0, :k2 => 0.66, :η1 => 2.0, :η2 => 2.0]), ImplicitEM(); saveat=1.0)
+    sol_1_2 = solve(SDEProblem(noise_scaling_network_1, u0, (0.0, 1000.0), [:k1 => 2.0, :k2 => 0.66, :η1 => 2.0, :η2 => 0.2]), ImplicitEM(); saveat=1.0)
+    sol_1_3 = solve(SDEProblem(noise_scaling_network_1, u0, (0.0, 1000.0), [:k1 => 2.0, :k2 => 0.66, :η1 => 0.2, :η2 => 0.2]), ImplicitEM(); saveat=1.0)
+    @test var(sol_1_1[1,:]) > var(sol_1_2[1,:]) > var(sol_1_3[1,:]) 
+
+    noise_scaling_network_2 = @reaction_network begin 
+        @noise_scaling_parameters η[1:2]
+        (k1, k2), X1 ↔ X2 
+    end
+    @unpack k1, k2, η = noise_scaling_network_2
+    sol_2_1 = solve(SDEProblem(noise_scaling_network_2, u0, (0.0, 1000.0), [k1 => 2.0, k2 => 0.66, η[1] => 2.0, η[2] => 2.0]), ImplicitEM(); saveat=1.0)
+    sol_2_2 = solve(SDEProblem(noise_scaling_network_2, u0, (0.0, 1000.0), [k1 => 2.0, k2 => 0.66, η[1] => 2.0, η[2] => 0.2]), ImplicitEM(); saveat=1.0)
+    sol_2_3 = solve(SDEProblem(noise_scaling_network_2, u0, (0.0, 1000.0), [k1 => 2.0, k2 => 0.66, η[1] => 0.2, η[2] => 0.2]), ImplicitEM(); saveat=1.0)
+    @test var(sol_2_1[1,:]) > var(sol_2_2[1,:]) > var(sol_2_3[1,:]) 
+end
+
```

Comment:
What are these covering that wouldn't be covered by an existing test?

Reply:
They cover that the amount of noise is actually changed when you change the noise scaling parameter. The second one checks that the feature works when symbolic are used rather than symbols. If you want I could remove the second case, although I figured it was better to have as well, but CI time hasn't really been a main concern, which might be unwise.

Reply:
Our CI is taking 26 minutes, see below, so generally I think we need to be mindful of not adding things that increase time while not providing new coverage. Should we instead just update the older noise scaling tests to use your new interface, maybe keeping one of them with the old style kwarg? That should be sufficient that it is propagating the values correctly. We already have tests that pass parameters to ODEs/SDEs, so I don't think we need new tests for that.

Moreover, given that these are SDEs, are these tests guaranteed to always pass? We don't want tests that can randomly fail due to poor luck with random numbers (and at a minimum we should be using `StableRNGs` for any new tests involving stochastic solvers).

Reply:
The number differences should be so huge that random fails shouldn't be possible but good point. I removed the previous noise scaling test as they were worse than the current one, and not testing anything different.

Good point on CI, yeah, I haven't really thought much about CI time, but should probably. If I have some extra time some time I will try to go through CI to figure out what is taking time, and if something can be cut.